src/coreclr/src/jit/morph.cpp

```diff
@@ -13912,19 +13956,42 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
                 // normalization.
                 if (temp->OperIs(GT_LCL_VAR))
                 {
-#ifdef DEBUG
-                    // We clear this flag on `temp` because `fgMorphLocalVar` may assert that this bit is clear
-                    // and the node in question must have this bit set (as it has already been morphed).
-                    temp->gtDebugFlags &= ~GTF_DEBUG_NODE_MORPHED;
-#endif // DEBUG
-                    const bool forceRemorph = true;
-                    temp                    = fgMorphLocalVar(temp, forceRemorph);
-#ifdef DEBUG
-                    // We then set this flag on `temp` because `fgMorhpLocalVar` may not set it itself, and the
-                    // caller of `fgMorphSmpOp` may assert that this flag is set on `temp` once this function
-                    // returns.
-                    temp->gtDebugFlags |= GTF_DEBUG_NODE_MORPHED;
-#endif // DEBUG
+                    unsigned  lclNum  = temp->AsLclVarCommon()->GetLclNum();
+                    var_types lclType = lvaTable[lclNum].lvType;
+
+                    // We build a new 'result' tree to return, as we want to call fgMorphTree on it
+                    //
+                    GenTree* result = gtNewLclvNode(lclNum, lclType);
+                    assert(result->OperGet() == GT_LCL_VAR);
```

Comment:
Why is it that we need to build a new result tree? It seems an unfortunate additional cost; why doesn't it work to clearn the `GTF_DEBUG_NODE_MORPHED` flag?

Reply:
The whole clearing/setting the GTF_DEBUG_NODE_MORPHED flag is a total hack.

But fine I will revert the change to my original fix which is much simpler but introduces some regressions.

I believe that this fix is better, but will go with whatever you want.\

Reply:
I totally agree that the `GTF_DEBUG_NODE_MOPRHED` flag is a frustratingly fragile mechanism - though I do think it would be best to do a minimal fix for now.